src/lib/solver.ts

```
@@ -0,0 +1,165 @@
+import type {Household, Person, Meal, Dish, ISODateString} from "$lib/cache.svelte"
+import {type Variable, EQ, TERMS, TERM, SUM, CONST, type LinearEquation} from "$lib/linear-algebra"
+import {type Nutrition, targetCalories, targetProtein, nutritionOfDish} from "$lib/nutrition"
+
+////////////////////////////////////////////////////////////////////////////////
+// SOLVER
+////////////////////////////////////////////////////////////////////////////////
+
+export function calculateHousehold(household: Household): Record<Person["id"], Record<Meal["id"], Record<Dish["id"], number>>> {
+	const models = makeModels(household)
+	const solution: Record<Person["id"], Record<Meal["id"], Record<Dish["id"], number>>> = {}
+	Object.entries(models).forEach(([day, dayModels]) => Object.entries(dayModels).forEach(([person, model]) => solution[Number(person)] = solveModel(model, day, Number(person))))
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)